data/modules/Equipment/Internal.lua

```diff
@@ -0,0 +1,242 @@
+	l10n_key="TRADE_COMPUTER",
+	price=400, purchasable=true, tech_level=9,
+	slot={ type="computer", size=1 },
+	mass=0.2, volume=0.5, capabilities={ trade_computer=1 },
+	icon_name="equip_trade_computer"
+})
+
+--===============================================
+-- Sensors
+--===============================================
+
+Equipment.Register("sensor.radar", SensorType.New {
+	l10n_key="RADAR",
+	price=680, purchasable=true, tech_level=3,
+	slot = { type="sensor.radar", size=1 },
+	mass=1.0, volume=1.0, capabilities = { radar=1 },
+	icon_name="equip_radar"
+})
+
+--===============================================
+-- Shield Generators
+--===============================================
+
+Equipment.Register("shield.basic_s1", EquipType.New {
+	l10n_key="SHIELD_GENERATOR",
+	price=2500, purchasable=true, tech_level=5,
+	slot = { type="shield", size=1 },
+	mass=1, volume=2, capabilities = { shield=1 },
+	icon_name="equip_shield_generator"
+})
+
+Equipment.Register("shield.basic_s2", EquipType.New {
+	l10n_key="SHIELD_GENERATOR",
+	price=5500, purchasable=true, tech_level=7,
+	slot = { type="shield", size=2 },
+	mass=2.8, volume=5, capabilities = { shield=2 },
+	icon_name="equip_shield_generator"
+})
+
+Equipment.Register("shield.basic_s3", EquipType.New {
+	l10n_key="SHIELD_GENERATOR",
+	price=11500, purchasable=true, tech_level=8,
+	slot = { type="shield", size=3 },
+	mass=7, volume=12.5, capabilities = { shield=3 },
+	icon_name="equip_shield_generator"
+})
+
+Equipment.Register("shield.basic_s4", EquipType.New {
+	l10n_key="SHIELD_GENERATOR",
+	price=23500, purchasable=true, tech_level=9,
+	slot = { type="shield", size=4 },
+	mass=17.9, volume=32, capabilities = { shield=4 },
+	icon_name="equip_shield_generator"
+})
+
+Equipment.Register("shield.basic_s5", EquipType.New {
+	l10n_key="SHIELD_GENERATOR",
+	price=58500, purchasable=true, tech_level=10,
+	slot = { type="shield", size=5 },
+	mass=43.7, volume=78, capabilities = { shield=5 },
+	icon_name="equip_shield_generator"
+})
+
+--===============================================
+-- Hull Modifications
+--===============================================
+
+Equipment.Register("hull.atmospheric_shielding", EquipType.New {
+	l10n_key="ATMOSPHERIC_SHIELDING",
+	price=200, purchasable=true, tech_level=3,
+	slot = { type="hull.atmo_shield", size=1 },
+	mass=1, volume=2, capabilities = { atmo_shield=9 },
+	icon_name="equip_atmo_shield_generator"
+})
+
+Equipment.Register("hull.heavy_atmospheric_shielding", EquipType.New {
+	l10n_key="ATMOSPHERIC_SHIELDING_HEAVY",
+	price=900, purchasable=true, tech_level=5,
+	slot = { type="hull.atmo_shield", size=3 },
+	mass=5, volume=12, capabilities = { atmo_shield=19 },
+	icon_name="equip_atmo_shield_generator"
+})
+
+Equipment.Register("misc.hull_autorepair", EquipType.New {
+	l10n_key="HULL_AUTOREPAIR", slots="hull_autorepair",
+	price=16000, purchasable=true, tech_level="MILITARY",
+	slot = { type="hull.autorepair", size=4 },
```

Comment:
Currently no ship has explicit Hull slots defined, so they all get a single S1 Hull slot.

This makes this equipment impossible to install.

Reply:
I'm peripherally aware of this already, see the TODO list at the top of the PR / the associated Github Project board for the work item regarding ship hull equipment.